### PR TITLE
FastbillAutomatic

### PR DESCRIPTION
**Pull request by awdng**
Some basic FastbillAutomatic Integration, does allow handling of Subscriptions.
Changed the structure a bit to have a standard MyFastbillClient and a FastbillAutomatic client which extends the MyFastbillClient because it has the same functionality and adds a few extra things like subscriptions.

Not all API features are supported, but it's a start.

### CODE REVIEW

**Comment by pscheit** [comment]
Hi,

thank you so much for this PR! I really appreciate your efforts. I just cloned it and tested it, and it looks well.
Please run

```
composer update webforge/testplate
```

and commit the changed composer.lock file. Use phpunit 4.4.x to run the tests.
- Please fix the codestyle for new classes and for the AbstractFastBillClient.
- the only BC break is that `Client` was renamed to `MyFastBillClient` ?

File: lib/FastBill/Api/AbstractFastBillClient.php [hunk 1]
```diff
@@ -41,18 +40,18 @@ public function __construct(GuzzleClient $guzzleClient, Array $options) {
    */
   public function createInvoice(Invoice $invoice) {
     $requestBody = array(
-      'SERVICE' => 'invoice.create',
-      'DATA' => $invoice->serializeJSONXML()
+        'SERVICE' => 'invoice.create',
```
Comment (pscheit) [inline lib/FastBill/Api/AbstractFastBillClient.php]: (+ the following)
indent with 2 spaces

File: lib/FastBill/Api/BadRequestException.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace FastBill\Api;
+
+class BadRequestException extends \RuntimeException {
+
+    protected $errors = array();
+
+    public static function fromResponse(\stdClass $jsonResponse)
+    {
```
Comment (pscheit) [inline lib/FastBill/Api/BadRequestException.php:11]: indent with 2 spaces. use k&r style for braces (curly brace on same line as function statement)

File: tests/FastBill/Api/ClientSubscriptionTest.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+use FastBill\Model\Subscription;
+use Webforge\Common\System\File;
+use Guzzle\Http\Message\Request as GuzzleRequest;
+use Webforge\Common\String as S;
+use Webforge\Common\Preg;
+
+class ClientSubscriptionTest extends \FastBill\Model\Test\ModelTestCase {
+  
+  public function setUp() {
+    $this->chainClass = __NAMESPACE__ . '\\Client';
```
Comment (pscheit) [inline tests/FastBill/Api/ClientSubscriptionTest.php]: you can remove this

**Comment by awdng** [comment]
Thanks for the feedback, will incorporate those changes. I have my IDE set to autodetect Code Styles, but as i worked on the package while integrated into another project with different Code Style, it did not seem to autodetect the style in this case.

That being said, any reason you don't follow PSR-2 ? I had the impression it's pretty much the standard in the PHP world now, and not using it just makes your life (and that of other devs) that tiny bit harder :)

**Comment by pscheit** [comment]
yeah I was to lazy to adapt then. I could live with every class from this project in PSR-2 (if you're willing to do that). But not mixed style :)

**Comment by awdng** [comment]
reformatted the files and updated the composer.lock file to reflect the testplate package update

**Pull request merged**